### PR TITLE
Don't re-check with gcloud for a firewall's existence to fix a potential flake.

### PR DESCRIPTION
**Pull request by brendandburns**
Ref: https://github.com/kubernetes/kubernetes/issues/20306

@ixdy @spxtr @ihmccreery

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -1265,8 +1265,12 @@ function test-setup {
   until gcloud compute firewall-rules describe --project "${PROJECT}" "${NODE_TAG}-${INSTANCE_PREFIX}-http-alt" 2> /dev/null || [ $(($start + 1200)) -lt `date +%s` ]
   do sleep 5
   done
-  # Check if the firewall rule exists and fail if it does not.
-  gcloud compute firewall-rules describe --project "${PROJECT}" "${NODE_TAG}-${INSTANCE_PREFIX}-http-alt"
+  # Note, determine if we timed out, or if we found the firewall, you might think we could just
+  # check with gcloud, but sometimes that call flakes.  https://github.com/kubernetes/kubernetes/issues/20306
+  if [ $(($start + 1200)) -lt `date +%s` ]; then
```
Comment (ixdy) [inline cluster/gce/util.sh]: can you make this timeout a constant and then use it here and in the previous loop?

Comment (ixdy) [inline cluster/gce/util.sh]: actually, maybe just rework the check entirely: replace the `until` on L1265 with something like
```bash
while ! gcloud compute firewall-rules describe --project "${PROJECT}" "${NODE_TAG}-${INSTANCE_PREFIX}-http-alt" 2> /dev/null; do
  if [[ $(($start + 1200)) -lt `date +%s` ]]; then
    echo -e "${color_red}Failed to create firewall ${NODE_TAG}-${INSTANCE_PREFIX}-http-alt in ${PROJECT}" >&2
    exit 1
  fi
  sleep 5
done

and similarly replace the loop to check the nodeports firewall just after this.

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -1265,8 +1265,12 @@ function test-setup {
...
+    echo "Failed to create a firewall ${NODE_TAG}-${INSTANCE_PREFIX}-http-alt in ${PROJECT}.
```
Comment (ixdy) [inline cluster/gce/util.sh]: note this is missing the final ".

should also probably use `${color_red}` and echo to stderr.

File: cluster/gce/util.sh (hunk 1 above)
Comment (brendandburns) [inline cluster/gce/util.sh]: done.

File: cluster/gce/util.sh (hunk 2 above)
Comment (brendandburns) [inline cluster/gce/util.sh]: done.

**Comment by brendandburns** [comment]
@ixdy comment addressed, ptal.

thanks!
--brendan

**Comment by ixdy** [comment]
LGTM

**Pull request merged**